### DISCUSSION TITLE
Minio Console can not be served on subpath using the latest Minio version

### DISCUSSION THREAD

**Discussion by siddjellali** [minio/minio#16304] [Q&A] [answered]
Hello, 

Subpath seems to be not working [#14761](https://github.com/minio/minio/pull/14761)

**nginx**
```
location /s3/ {
	rewrite ^/s3/(.*)$ $1 break;
	proxy_pass http://localhost:9001/s3/;
	expires -1;
 }
```

**Dockerfile**
```
ENV CONSOLE_SUBPATH /s3/
ENV MINIO_BROWSER_REDIRECT_URL http://localhost:9001/s3
```
`Version: DEVELOPMENT.2022-10-08T20-11-00Z`

![image](https://user-images.githubusercontent.com/3965823/195706680-b56dd7b9-68a2-4895-9a64-57967aea69a2.png)

BASE HTML balise seems good...

![image](https://user-images.githubusercontent.com/3965823/195706833-b27968b6-2945-42f7-9e91-79d84ef8e3fd.png)

Thanks for your support.

**Comment by harshavardhana** [comment]
> `Version: DEVELOPMENT.2022-10-08T20-11-00Z`

You are compiling your own ? 

> ENV CONSOLE_SUBPATH /s3/

This is useless to set

> location /s3/ {
> 	rewrite ^/s3/(.*)$ $1 break;
> 	proxy_pass http://localhost:9001/s3/;
> 	expires -1;
>  }

This is wrong https://github.com/minio/minio/pull/14761#issue-1205981691

Correct

```
location /s3/ {
	rewrite ^/s3/(.*)$ $1 break;
	proxy_pass http://localhost:9001/;
	expires -1;
 }
```

MINIO_BROWSER_REDIRECT_URL="http://localhost:8000/s3"  for starting MinIO

**Comment by siddjellali** [comment]
Excellent, thanks a lot.
Do you know if it's possible to have the same with API as well ?

**Comment by harshavardhana** [comment]
> Do you know if it's possible to have the same with API as well ?

No that is not allowed in S3 API

**Comment by carlvine500** [comment, marked as answer]
1. nginx.conf

```
    server {
        listen 7077;
        location /minio/ {
          rewrite "^/minio/(.*)$" /$1 break;
          proxy_pass http://localhost:9001;
        }
    }
```
2. docker command:
```
 docker run -p  9000:9000 -d -p 9001:9001 -e "MINIO_BROWSER_REDIRECT_URL=http://localhost:9001/minio/" -e "MINIO_ROOT_USER=minio111" -e "MINIO_ROOT_PASSWORD=minio111" quay.io/minio/minio server /data --console-address ":9001"
```
3. browser  url: http://192.168.1.14:7077/minio/login


![1211671780629_ pic](https://user-images.githubusercontent.com/7055460/209292226-e2e9f322-7951-4420-8f5c-d5aeee0625e7.jpg)

**Reply by tassoman** [reply]
Did you checked websocket is fully functioning? I'm getting `400 bad request` error back

[ EDIT ] 
I had to upgrade the connection:

```
        location /minio/ {
                rewrite "^/minio/(.*)$" /$1 break;
                proxy_pass http://localhost:9001;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection "Upgrade";
        }
```

But now the web service is giving error **403** :thinking:

**Comment by mqu** [comment]
I have build a fully working example, based on many sources with an embed NGINX frontend : https://github.com/mqu/minio-docker-compose-console

- minio URL : http://localhost:8888/ 
- I have been able to add an apache frontend as usal for many dockerized stack application on my production VM.

best regards, Marc.

**Reply by tallandtree** [reply]
Thanks, I was breaking my head over this change until I found this discussion. I couldn't get the console working fully with the 400 Bad Request when I updated from `RELEASE.2022-12-02T19-19-22Z` to `RELEASE.2022-12-07T00-56-37Z`. With the 

```
   proxy_set_header Upgrade $http_upgrade;
   proxy_set_header Connection "upgrade";
```
lines added, it worked again.

**Comment by carlvine500** [comment]
this version works ok too:
 docker run -p  9000:9000 -d -p 9001:9001 -e "MINIO_BROWSER_REDIRECT_URL=http://localhost:9001/minio/"  -e "MINIO_ROOT_USER=minio123" -e "MINIO_ROOT_PASSWORD=minio123" quay.io/minio/minio:RELEASE.2022-05-08T23-50-31Z server /data --console-address ":9001"
